filament/src/Renderer.cpp

```diff
@@ -1171,6 +1173,10 @@ void FRenderer::endFrame() {
 
 void FRenderer::readPixels(uint32_t xoffset, uint32_t yoffset, uint32_t width, uint32_t height,
         PixelBufferDescriptor&& buffer) {
+#ifndef NDEBUG
```

Comment:
Is there anything useful we could do in release build?

Reply:
The only thing we could do is skip the call (and call the callback). I'm against logging something because it can quickly become log spam, unless we have a way to log just once.

Reply:
I was also wondering if we could change the API to return true/false for instance or something like this. I'm afraid just doing a no-op would lead to annoying bug reports.

Reply:
It would be a bit weird because basically the return value would be completely determined by whether it's called issue l inside begin/end frame. So it's not very different from just documenting it.

Also if we return an error, what do we do with the callback? 

It would be weird to call it if there was a reported error, on the other hand, if we don't, we could leak memory.

Usually filament silently fails when you don't use it properly, I think it should be the same here. Logging something just once could be nice.

Reply:
Note that `readPixels` in release builds will continue to work as before. The `NDEBUG` check is just there to mitigate risk and can be removed in the future.